File: Frontend/src/types/technicalVisit.ts
export type TechnicalVisitResponse = {
  id: string
  vehicleId: string
  vehiclePlate: string
  vehicleLabel: string
  type: string
  scheduledDate: string
  completedDate: string | null
  result: string
  garage: string | null
  cost: number | string | null
  nextDueDate: string | null
  comments: string | null
  createdAt: string
}

export type TechnicalVisitPageResponse = {
  content: TechnicalVisitResponse[]
  page: number
  size: number
  totalElements: number
  totalPages: number
}

export type CreateTechnicalVisitPayload = {
  vehicleId: string
  type: string
  scheduledDate: string
  completedDate?: string | null
  result?: string | null
  garage?: string | null
  cost?: number | string | null
  nextDueDate?: string | null
  comments?: string | null
}

export type UpdateTechnicalVisitPayload = {
  type: string
  scheduledDate?: string | null
  completedDate?: string | null
  result?: string | null
  garage?: string | null
  cost?: number | string | null
  nextDueDate?: string | null
  comments?: string | null
}

File: Backend/src/main/java/com/leader/parcautomobile/entity/TechnicalVisit.java
package com.leader.parcautomobile.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "technical_visits")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TechnicalVisit {

	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	private UUID id;

	@ManyToOne(optional = false, fetch = FetchType.LAZY)
	@JoinColumn(name = "vehicle_id", nullable = false)
	private Vehicle vehicle;

	@Column(nullable = false, length = 50)
	private String type;

	@Column(name = "scheduled_date", nullable = false)
	private LocalDate scheduledDate;

	@Column(name = "completed_date")
	private LocalDate completedDate;

	@Column(length = 30)
	@Builder.Default
	private String result = "EN_ATTENTE";

	@Column(length = 200)
	private String garage;

	@Column(precision = 12, scale = 2)
	private BigDecimal cost;

	@Column(name = "next_due_date")
	private LocalDate nextDueDate;

	@Column(columnDefinition = "text")
	private String comments;

	@Column(name = "created_at", nullable = false)
	private Instant createdAt;

	@PrePersist
	void prePersist() {
		if (createdAt == null) {
			createdAt = Instant.now();
		}
	}
}

